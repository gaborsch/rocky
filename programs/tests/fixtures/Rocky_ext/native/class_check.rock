from java, math play BigDecimal
from java, lang play Number, Comparable
from java, io play Serializable  

X will be BigDecimal taking "1.23"

if X is like BigDecimal
  Say "BigDeciml (runtime class) OK"
else
  Say "BigDeciml (runtime class) Failed"

if X is like Number
  Say "Number (parent class) OK"
else
  Say "Number (parent class) Failed"

if X is like Comparable
  Say "Comparable (interface) OK"
else
  Say "Comparable (interface) Failed"
  
if X is like Serializable
  Say "Serializable (parent's interface) OK"
else
  Say "Serializable (parent's interface) Failed"
  
Y will be BigDecimal taking 1.23
if X is like Y
  Say "Y (another instance) OK"
else
  Say "Y (another instance) Failed"

if BigDecimal is like Number
  Say "BigDecimal-Number (class type) OK"
else
  Say "BigDecimal-Number (class type) Failed"

if Number is like BigDecimal
  Say "Number-BigDecimal (inverse) Failed"
else
  Say "Number-BigDecimal (inverse) OK"


if BigDecimal is not like Number
  Say "BigDecimal-Number (negated) Failed"
else
  Say "BigDecimal-Number (negated) OK"

