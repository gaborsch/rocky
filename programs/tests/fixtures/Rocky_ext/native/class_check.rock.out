BigDeciml (runtime class) OK
Number (parent class) OK
Comparable (interface) OK
Serializable (parent's interface) OK
Y (another instance) OK
BigDecimal-Number (class type) OK
Number-BigDecimal (inverse) OK
BigDecimal-Number (negated) OK
